specification/oracle/models/exadata/exadatainfra.tsp

```diff
@@ -0,0 +1,269 @@
+  @doc("Indicates that resource in Provisioning state")
+  Provisioning: "Provisioning",
+
+  @doc("Indicates that resource in Available state")
+  Available: "Available",
+
+  @doc("Indicates that resource in Updating state")
+  Updating: "Updating",
+
+  @doc("Indicates that resource in Terminating state")
+  Terminating: "Terminating",
+
+  @doc("Indicates that resource in Terminated state")
+  Terminated: "Terminated",
+
+  @doc("Indicates that resource maintenance in progress state")
+  MaintenanceInProgress: "MaintenanceInProgress",
+
+  @doc("Indicates that resource in Failed state")
+  Failed: "Failed",
+
+  string,
+}
+
+@doc("MaintenanceWindow resource properties")
+model MaintenanceWindow {
+  @doc("The maintenance window scheduling preference.")
+  preference?: Preference;
+
+  @doc("Months during the year when maintenance should be performed.")
+  @extension("x-ms-identifiers", ["name"])
+  months?: Month[];
+
+  @doc("Weeks during the month when maintenance should be performed. Weeks start on the 1st, 8th, 15th, and 22nd days of the month, and have a duration of 7 days. Weeks start and end based on calendar dates, not days of the week. For example, to allow maintenance during the 2nd week of the month (from the 8th day to the 14th day of the month), use the value 2. Maintenance cannot be scheduled for the fifth week of months that contain more than 28 days. Note that this parameter works in conjunction with the  daysOfWeek and hoursOfDay parameters to allow you to specify specific days of the week and hours that maintenance will be performed. ")
+  weeksOfMonth?: int32[];
+
+  @doc("Days during the week when maintenance should be performed.")
+  @extension("x-ms-identifiers", ["name"])
+  daysOfWeek?: DayOfWeek[];
+
+  @doc("The window of hours during the day when maintenance should be performed. The window is a 4 hour slot. Valid values are - 0 - represents time slot 0:00 - 3:59 UTC - 4 - represents time slot 4:00 - 7:59 UTC - 8 - represents time slot 8:00 - 11:59 UTC - 12 - represents time slot 12:00 - 15:59 UTC - 16 - represents time slot 16:00 - 19:59 UTC - 20 - represents time slot 20:00 - 23:59 UTC")
+  hoursOfDay?: int32[];
+
+  @doc("Lead time window allows user to set a lead time to prepare for a down time. The lead time is in weeks and valid value is between 1 to 4. ")
+  leadTimeInWeeks?: int32;
+
+  @doc("Cloud Exadata infrastructure node patching method.")
+  patchingMode?: PatchingMode;
+
+  // OCI Docs say min value is 15, but when making the call in OCI, observed a value of 0, so setting minimum value to 0.
+  // Link: https://docs.oracle.com/en-us/iaas/api/#/en/database/20160918/datatypes/MaintenanceWindow
+  @doc("Determines the amount of time the system will wait before the start of each database server patching operation. Custom action timeout is in minutes and valid value is between 15 to 120 (inclusive).")
+  @minValue(0)
+  @maxValue(120)
+  customActionTimeoutInMins?: int32;
```

Comment:
if the range is from 0-120, maybe uint8 is more appropriate type here, not sure though.

Reply:
for uint8 we got an error: Model type 'uint8' is not supported in Azure resource manager APIs.